jvm-libs/linea/clients/eth-logs-searcher/src/test/kotlin/linea/ethapi/LogsSearcherTest.kt

```diff
@@ -0,0 +1,159 @@
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import kotlin.time.Duration.Companion.hours
+import kotlin.time.Duration.Companion.seconds
+
+class LogsSearcherTest {
+  private val testAddress = "0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa01"
+  private val testTopic1 = "0x1111111111111111111111111111111111111111111111111111111111111111"
+  private val testTopic2 = "0x2222222222222222222222222222222222222222222222222222222222222222"
+  private val templateLog = EthLog(
+    blockNumber = 100UL,
+    address = testAddress.decodeHex(),
+    topics = emptyList(),
+    data = "0x1234567890abcdef".decodeHex(),
+    transactionHash = "0xabcdefabcdefabcdefabcdefabcdefabcdefabcdef".decodeHex(),
+    transactionIndex = 0UL,
+    logIndex = 0UL,
+    blockHash = "0xabcdefabcdefabcdefabcdefabcdefabcdefabcdef".decodeHex(),
+    removed = false
+  )
+  private val initialLogs = listOf(
+    templateLog.copy(
+      blockNumber = 200UL,
+      topics = listOf(testTopic1.decodeHex(), testTopic2.decodeHex())
+    ),
+    templateLog.copy(
+      blockNumber = 300UL,
+      topics = listOf(testTopic1.decodeHex())
+    ),
+    templateLog.copy(
+      blockNumber = 400UL,
+      address = "0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa02".decodeHex(),
+      topics = listOf(testTopic2.decodeHex())
+    )
+  )
+  private lateinit var vertx: Vertx
+  private lateinit var fakeElClient: FakeEthApiClient
+  private lateinit var searcher: EthLogsSearcher
+
+  @BeforeEach
+  fun setUp() {
+    vertx = Vertx.vertx()
+    fakeElClient = FakeEthApiClient(initialLogs.toSet())
+    fakeElClient.setLatestBlockTag(initialLogs.last().blockNumber + 1UL)
+    searcher = EthLogsSearcherImpl(vertx, fakeElClient)
+  }
+
+  @AfterEach
+  fun tearDown() {
+    vertx.close()
+  }
+
+  @Test
+  fun `should return logs within block range`() {
+    fakeElClient.setFinalizedBlockTag(450UL)
+    val result = searcher.getLogsRollingForward(
+      fromBlock = BlockParameter.BlockNumber(10UL),
+      toBlock = BlockParameter.Tag.FINALIZED,
+      address = testAddress,
+      topics = emptyList(),
+      chunkSize = 10U,
+      searchTimeout = 1000000.seconds,
+      stopAfterTargetLogsCount = null
+    ).get()
+
+    assertThat(result.logs).isEqualTo(initialLogs.take(2))
+    assertThat(result.startBlockNumber).isEqualTo(10UL)
+    assertThat(result.endBlockNumber).isEqualTo(450UL)
+  }
+
+  @Test
+  fun `should filter logs by address`() {
+    val result = searcher.getLogsRollingForward(
+      fromBlock = BlockParameter.BlockNumber(0UL),
+      toBlock = BlockParameter.BlockNumber(500UL),
+      address = "0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa02",
+      topics = emptyList(),
+      chunkSize = 100U,
+      searchTimeout = 5.seconds,
+      stopAfterTargetLogsCount = null
+    ).get()
+
+    assertThat(result.logs).isEqualTo(listOf(initialLogs[2]))
+  }
+
+  @Test
+  fun `should filter logs by topics`() {
```

Comment:
Consider adding more logs with `testTopic2` and `testAddress`, because this test is indistinguishable from the previous one, that is testing filtering by address